src/lexer/tiger.lex

```diff
@@ -1,7 +1,10 @@
 type pos = int
 type lexresult = Token.token
 
-fun eof() = Token.EOF
+fun eof() =
+    (if SrcString.isBuildingString() then raise SrcString.StringNotClosed(Newline.getPos(SrcString.getStartPos()), Newline.getLine(SrcString.getStartPos()))
+    else if not (SrcComment.closed()) then raise SrcComment.CommentNotClosed(Newline.getPos(hd(SrcComment.getComments())), Newline.getLine(hd(SrcComment.getComments())))
```

Comment:
We can talk about this more before or after class, but wouldn't we want the compiler to just exit since we know we can't move on to the parser?